addons/website/static/src/components/dialog/duplicate_object.js

```diff
@@ -0,0 +1,57 @@
+/** @odoo-module **/
+
+import {useService, useAutofocus} from '@web/core/utils/hooks';
+
+import {WebsiteDialog} from './dialog';
+const {Component, useState, onWillStart} = owl;
+import {_t} from 'web.core';
+import wUtils from 'website.utils';
+
+export class DuplicateObjectDialog extends Component {
+    setup() {
+        this.orm = useService('orm');
+        this.website = useService('website');
+        this.notification = useService('notification');
+        useAutofocus();
+        this.state = useState({
+            name: '',
+            modalTitle: '',
+            objectName: '',
+        });
+        this.title = _t('Duplicate');
+        onWillStart(this._updateModalTitle);
+    }
+    async duplicate() {
+        if (!this.state.name) {
+            this.notification.add(_t('Please enter a name.'), {
+                title: _t('Attention'),
+                type: 'warning',
+            });
+            return;
+        }
+        const mainObject = this.website.currentWebsite.metadata.mainObject;
+        if (mainObject.model === 'website.page') {
+            const res = await this.orm.call(
+                mainObject.model,
+                'clone_page',
+                [mainObject.id, this.state.name]
+            );
+            this.website.goToWebsite({path: res, edition: true});
+        } else {
+            wUtils.sendRequest('/website/duplicate', {
```

Comment:
With that, the whole webclient will be reloaded to the duplicated object url (displayed in the client action), but it would be better to stay on the webclient and launch an action when receiving the response:
The `/website/duplicate` controller could probably be defined as a `type="json"` one and return the url (without enable_editor), 
In the component, the rpc service could be used instead of `wUtils.sendRequest`: 
`const url = await this.rpc('/website/duplicate', ...)`
After, the client action can be redirected with that url using goToWebsite :)

Reply:
Good one, thanks :+1: